R/get_regional_data.R

```diff
@@ -2,38 +2,45 @@
 #'
 #' @description Downloads, cleans and processes regional data
 #' for Covid-19.
-#' @inheritParams new_covidregionaldata
-#' @inheritParams process_regional
-#' @inheritParams return_regional
-#' @param ... pass additional arguments to `download_regional`
+#' @param country A character string specifying the country to get data from.
+#' Not case dependent. Name should be the English name. For a list of
+#' options see `available_datasets`.
+#' @param level A character string indicating the target administrative level
+#' of the data with the default being "1". Currently supported options are
+#' level 1 ("1) and level 2 ("2"). See `available_datasets` for supported
+#' options by dataset.
+#' @param totals Logical, defaults to FALSE. If TRUE, returns totalled
+#'  data per region up to today's date. If FALSE, returns the full dataset
+#'  stratified by date and region.
+#' @param localise Logical, defaults to TRUE. Should region names be localised.
+#' @param verbose Logical, defaults to `TRUE`. Should verbose processing
+#' messages and warnings be returned.
+#' @param steps Logical, defaults to FALSE. Should all processing and cleaning
+#' steps be kept and output in a list.
+#' @param ... additional arguments to pass to Country classes.
 #' @return A tibble with data related to cases, deaths, hospitalisations,
 #' recoveries and testing stratified by regions within the given country.
 #' @export
-#' @examples
-#' \dontrun{
-#' # set up a data cache
-#' start_using_memoise()
-#'
-#' # download the data (and keep all processing steps)
-#' get_regional_data(country = "Mexico", steps = TRUE)
-#' }
 #'
 get_regional_data <- function(country, level = "1", totals = FALSE,
                               localise = TRUE, verbose = TRUE,
                               steps = FALSE, ...) {
-  # check data availability and define list
-  region <- new_covidregionaldata(country, level = level, verbose = verbose)
+  # check data availability and initiate country class if avaliable
+  region_class <- check_country_avaliable(
+    country = country, level = level,
+    totals = totals, localise = localise,
+    verbose = verbose, steps = steps, ...
+  )
 
   # download and cache raw data
-  region <- download_regional(region, verbose = verbose, ...)
+  region_class$download_data()
 
   # dataset specifc cleaning
-  region <- clean_regional(region, verbose = verbose)
+  region_class$clean_regional()
 
   # non-specific cleaning and checks
-  region <- process_regional(region, totals = totals, 
-                             localise = localise, verbose = verbose)
+  region_class$process_regional()
 
-  region <- return_regional(region, steps = steps)
+  region <- region_class$return_regional()
```

Comment:
It seems weird to have S3 classes attached here? I'd like to be able to see the R6 object returned.

Reply:
This is so `get_regional_data` (and also `get_national_data`) returns a data frame of the final processed data (region$processed). If called with steps = True it returns the 'region' attribute of the R6 class, so raw and processed and others can be accessed as list elements. If we have it return the R6 class then the user would have to access the data with Country$region$processed (where Country is the country class name).

Reply:
fair enough - Seems pretty reasonable. We could consider generalising steps to be either give you just the processed data, the list of all data, or the underlying R6 object? Perhaps that is to complicated though (again can push to an issue and address after merging). Why does it have the S3 classes tagged on it still?

Reply:
TIm not sure, could it be because `self$region` is defined a structure in `dataClass` in `shared_methods`? I have made an issue for this and added it to the board. #155

Reply:
lets clean it up later then! :)

Reply:
Ah I think can just chuck these few lines that assign the attributes: https://github.com/epiforecasts/covidregionaldata/blob/411b6658f36cd34ce8ca7a3ae851262ce1583d09/R/shared-methods.R#L103

Reply:
Removed the class part in latest commit.